pkg/skaffold/docker/image.go

```diff
@@ -218,6 +218,24 @@ func (l *localDaemon) Push(ctx context.Context, out io.Writer, ref string) (stri
 		return "", errors.Wrapf(err, "getting auth config for %s", ref)
 	}
 
+	// Check if the local image has a digest from a previous push
+	// If the remote image has the same digest, there's no need
+	// to push.
+	image, _, err := l.apiClient.ImageInspectWithRaw(ctx, ref)
+	if err == nil {
+		if len(image.RepoDigests) == 1 {
+			parsed, err := ParseReference(image.RepoDigests[0])
```

Comment:
refactor this as `isAlreadyPushed`. 
```suggestion
			if parsed, ok := isAlreadyPushed(image.RepoDigests[0]); ok {
			   return parsed.Digest, nil
			}
```

Reply:
done